# FIN 4000 — Chapter 3: How Securities Are Traded
*Buying on margin, short selling, and the margin call that ties them together*

**ANSWER KEY — fully worked, instructor reference**

This notebook is the Python companion to the Chapter 3 Excel template — same data, same formulas, same numbers. Where the Excel workshop has you build a formula in a cell, this notebook has you build it in a line of Python instead.

## Step 1 — Entry Quiz *(3 minutes, individually)*

1. What's the difference between a market order and a limit order?
2. When you buy on margin, why does your percentage equity fall faster than the stock price itself does?
3. What does it mean to sell a stock short, and what is the short seller hoping will happen to the price?

---

In [1]:
import numpy as np

## Step 2 — Worked Example *(10 minutes, instructor-led)*

You buy 1,000 shares at $50 using 50% initial margin. Find the price that triggers a margin call.

In [2]:
shares = 1000
price = 50
initial_margin = 0.50
maintenance_margin = 0.30

In [3]:
position_value = shares * price
initial_equity = position_value * initial_margin
loan = position_value - initial_equity
margin_call_price = loan / (shares * (1 - maintenance_margin))
print(f'Loan=${loan:,.0f}  Margin call price=${margin_call_price:.2f}')

Loan=$25,000  Margin call price=$35.71


*Margin call price = $25,000 ÷ [1,000 × 0.70] = $35.71. The loan is fixed — only the stock's value moves — so as the price falls, that fixed loan becomes a bigger share of a shrinking pie.*

## Step 3 — Pair Work *(12 minutes, with a partner)*

**1.** The stock instead rises 20% to $60. Compute your new equity and your return on the $25,000 you invested — compare that to the stock's own 20% gain.

In [4]:
new_price = 60
new_position = shares * new_price
new_equity = new_position - loan
return_on_equity = (new_equity - initial_equity) / initial_equity
print(f'New equity=${new_equity:,.0f}  Return on equity={return_on_equity:.2%}')

New equity=$35,000  Return on equity=40.00%


**2.** The stock instead falls 20% to $40. Compute your new equity and your return. What does comparing the two cases tell you about leverage?

In [5]:
new_price_down = 40
new_position_down = shares * new_price_down
new_equity_down = new_position_down - loan
return_on_equity_down = (new_equity_down - initial_equity) / initial_equity
print(f'New equity=${new_equity_down:,.0f}  Return on equity={return_on_equity_down:.2%}')

New equity=$15,000  Return on equity=-40.00%


**3.** Now short sell the same way: 1,000 shares at $50, 50% initial margin, 30% maintenance margin. The $50,000 proceeds plus your $25,000 margin deposit sit in your account as collateral. At what price do you get a margin call?

In [6]:
proceeds = shares * price
initial_margin_posted = proceeds * initial_margin
total_account = proceeds + initial_margin_posted
margin_call_price_short = total_account / (shares * (1 + maintenance_margin))
print(f'Short-sale margin call price=${margin_call_price_short:.2f}')

Short-sale margin call price=$57.69


## Step 4 — Python Pass *(7 minutes)*

- The margin-call price formula for the long position, built from Loan/Shares/maintenance-margin cells
- The leveraged-return comparison at +20% and −20%, so the amplification is visible side by side
- The parallel margin-call price formula for the short position

In [7]:
def margin_call_price(shares, price, initial_margin, maintenance_margin, position='long'):
    if position == 'long':
        loan = shares * price * (1 - initial_margin)
        return loan / (shares * (1 - maintenance_margin))
    else:
        proceeds = shares * price
        collateral = proceeds * (1 + initial_margin)
        return collateral / (shares * (1 + maintenance_margin))

print(margin_call_price(1000, 50, 0.5, 0.3, 'long'))
print(margin_call_price(1000, 50, 0.5, 0.3, 'short'))

35.714285714285715
57.69230769230769


## Step 5 — Discussion *(3-5 minutes, no calculation)*

> A margin buyer gets a margin call when the price falls. A short seller gets a margin call when the price rises. Why are these opposite directions, and what does that tell you about how each position is exposed to risk?